collector/collector.go

```diff
@@ -152,26 +162,45 @@ func (e Exporter) Collect(ch chan<- prometheus.Metric) {
 	client := chrony.Client{Sequence: 1, Connection: conn}
 
 	if e.collectSources {
-		err = e.getSourcesMetrics(ch, client)
+		err = e.getSourcesMetrics(logger, ch, client)
 		if err != nil {
-			level.Debug(e.logger).Log("msg", "Couldn't get sources", "err", err)
+			level.Debug(logger).Log("msg", "Couldn't get sources", "err", err)
 			up = 0
 		}
 	}
 
 	if e.collectTracking {
-		err = e.getTrackingMetrics(ch, client)
+		err = e.getTrackingMetrics(logger, ch, client)
 		if err != nil {
-			level.Debug(e.logger).Log("msg", "Couldn't get tracking", "err", err)
+			level.Debug(logger).Log("msg", "Couldn't get tracking", "err", err)
 			up = 0
 		}
 	}
 
 	if e.collectServerstats {
-		err = e.getServerstatsMetrics(ch, client)
+		err = e.getServerstatsMetrics(logger, ch, client)
 		if err != nil {
-			level.Debug(e.logger).Log("msg", "Couldn't get serverstats", "err", err)
+			level.Debug(logger).Log("msg", "Couldn't get serverstats", "err", err)
 			up = 0
 		}
 	}
 }
+
+func (e Exporter) dnsLookup(logger log.Logger, address net.IP) string {
+	start := time.Now()
+	defer func() {
+		level.Debug(logger).Log("msg", "DNS lookup took", "seconds", time.Since(start).Seconds())
+	}()
+	if !e.dnsLookups {
+		return address.String()
+	}
+	names, err := net.LookupAddr(address.String())
```

Comment:
How about adding some caching capability here?

This is flooding our local DNS Server with several PTR requests every minute 😅

Reply:
Caching is something best handled by local system design. Something like CoreDNS can handle tens of thousands of requests per second. So, "several PTR requests every minute" is no problem.

Reply:
Yeah, the number of requests is no problem, the problem is that the amount of requests is _cluttering_ the local server _logs_, making hard trying to keep track of the requests we care about.

Thanks for the fast reply, btw.